services/prometheus/alert_manager.go

```diff
@@ -0,0 +1,83 @@
+package prometheus
+
+import (
+	"bytes"
+	"context"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"strings"
+	"time"
+
+	"github.com/percona/pmm/utils/pdeathsig"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	"golang.org/x/sys/unix"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+type AlertManager struct {
+	// To make this testeable. To run API tests we need to write the rules file but on dev envs
+	// there is no /srv/prometheus/rules/ directory
+	alertManagerFile string
+	l                *logrus.Entry
+}
+
+func NewAlertManager(alertManagerFile string) *AlertManager {
+	return &AlertManager{
+		alertManagerFile: alertManagerFile,
+		l:                logrus.WithField("component", "alert_manager"),
+	}
+}
+
+func (s *AlertManager) ValidateRules(ctx context.Context, rules string) error {
+	tempFile, err := ioutil.TempFile("", "temp_rules_*.yml")
+	if err != nil {
+		return errors.WithStack(err)
+	}
+	tempFile.Close()                 //nolint:errcheck
+	defer os.Remove(tempFile.Name()) //nolint:errcheck
+
+	if err = ioutil.WriteFile(tempFile.Name(), []byte(rules), 0644); err != nil {
```

Comment:
Why not use `tempFile.Write()` instead? The file already created and opened a few lines above.